oaknational/oak-ai-lesson-assistant
# test: record and replay streaming LLM calls in e2e tests

## Description

- Playwright intercepts API calls to `/chat` and adds `x-fixture-name` and `x-fixture-mode` headers
- Chat API handler looks for those headers and passes a custom LLMService to Aila
  - The FixtureRecord LLMService proxies the OpenAI service and saves the chunks to a file
  - The FixtureReplay LLMService streams the chunks in a fixture file


We aren't yet replaying moderation, RAG, or categorisation calls. I will add those in a follow up PR

## How to test

1. Check out the branch locally
2. Pull doppler, or set `AILA_FIXTURES_ENABLED=true` in `.env`
3. Run `pnpm run test-e2e-ui`
4. Click through to aila-chat => full-romans.test.ts => Authenticated => "Full aila flow with Romans fixture"
5. Run the test. If you make sure it's selected in the sidebar you should see the output on the right. Note that the auth setup will run first
6. You should see the same content as the fixtures. Look at the `.formatted.txt` files to confirm
